userapps/devshell/sysenter.h

```diff
@@ -19,9 +19,13 @@ do_sysenter_call0(int syscall)
                 "sysenter\n\t"
                 "1:\n\t"
                 : "=a" (ret)
-                : "a" (syscall)
+                : "a" (sysno)
                 : "memory", "cc");
 
+#elif defined(__riscv)
+
+   ret = syscall(sysno);
```

Comment:
Technically, I added those inline wrappers to test specifically the `sysenter` interface, which is not what is used by `syscall()`. Other time, I used them also for non-sysenter tests and so they stick. You don't need to change anything now. I just wanted to share with you the story behind that.